3scale/porta
# Upgrade pisoni to fix special characters in application ID

**What this PR does / why we need it**:

Fixes the issue where application keys are managed for applications (in App ID/App Key or OAuth auth mode) that have a special character in the application ID.

**Which issue(s) this PR fixes** 

https://issues.redhat.com/browse/THREESCALE-9033

**Verification steps** 

1. Set up App ID / App Key authentication for the service.
2. Create an application with special characters in application ID.
3. Go to the application page and try generating a random key, setting a custom key and deleting a key. All should be successful.

**Special notes for your reviewer**:

I tried adding an integration test, but there is an issue - all `ThreeScale::Core` calls are mocked, and I am not sure how to easily "unmock" the affected calls :shrug: (`ThreeScale::Core::ApplicationKey.save` and `ThreeScale::Core::ApplicationKey.delete`).
